### ISSUE TITLE
System java versions not preferred over java downloader

### ISSUE THREAD

**Issue by theofficialgman** [PrismLauncher/PrismLauncher#2809]
### Operating System

Linux

### Version of Prism Launcher

Develop (latest at time of post)

### Version of Qt

5.15.2

### Description of bug

When the automatic java downloader is used when launching an instance (eg: when the global selected java version is NOT the correct version for the version of minecraft, such as global java 21 with minecraft 1.8.9) the already installed system java versions are not used even if they match the version of java that is needed.

### Steps to reproduce

1. Install appropriate system java for the version of minecraft as well as another version of java
2. Select the global settings java to be one of the two versions
3. Launch a version of minecraft where the version of java needed is not the one in the global settings
4. Observe java is downloaded from online instead of the version of java available locally that is approriate

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by Trial97** [comment]
That is intended: when auto download java is enabled prism will always download if not already downloaded the recommended mojang version

**Comment by theofficialgman** [comment]
> That is intended: when auto download java is enabled prism will always download if not already downloaded the recommended mojang version

why? I have the same java version (even minor version) already installed to the mojang version. IMHO, if those match then it should just use the system one.

**Comment by Trial97** [comment]
Is hard to actually match that version:  the mapping from mojang only gives us the aliases for the java to use so we would need to match with a exact version and from what I saw mojang tends to ship different versions of java depending of os.
Then we have the part where we get to ask the user installed java what version is to match it against all that.

In the initial phase of Java stuff, I wanted only the basic stuff not go for each case.
I see your point but in your exact case just stick without the auto-download and only auto-detect.
PS. PRs are welcomed

**Comment by theofficialgman** [comment]
Fair enough. In that case this issue can be marked as an enhancement for now unless someone else feels strongly one way or another.
